core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java

```diff
@@ -149,6 +166,65 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static MultiGetResponse fromXContent(XContentParser parser) throws IOException {
+        String currentFieldName = null;
+        List<MultiGetItemResponse> items = new ArrayList<>();
+        for (Token token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+            switch (token) {
+                case FIELD_NAME:
+                    currentFieldName = parser.currentName();
+                    break;
+                case START_ARRAY:
+                    if (Fields.DOCS.equals(currentFieldName)) {
+                        for (token = parser.nextToken(); token != Token.END_ARRAY; token = parser.nextToken()) {
+                            if (token == Token.START_OBJECT) {
+                                String index = null;
+                                String type = null;
+                                String id = null;
+                                ElasticsearchException exception = null;
+                                GetResult getResult = null;
+                                item: for (token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+                                    switch (token) {
+                                        case FIELD_NAME:
+                                            currentFieldName = parser.currentName();
+                                            if (INDEX.match(currentFieldName) == false && TYPE.match(currentFieldName) == false &&
+                                                    ID.match(currentFieldName) == false && ERROR.match(currentFieldName) == false) {
+                                                getResult = GetResult.fromXContentEmbedded(parser, index, type, id);
+                                                break item;
+                                            }
+                                            break;
+                                        case VALUE_STRING:
+                                            if (INDEX.match(currentFieldName)) {
+                                                index = parser.text();
+                                            } else if (TYPE.match(currentFieldName)) {
+                                                type = parser.text();
+                                            } else if (ID.match(currentFieldName)) {
+                                                id = parser.text();
+                                            }
+                                            break;
+                                        case START_OBJECT:
+                                            if (ERROR.match(currentFieldName)) {
+                                                exception = ElasticsearchException.fromXContent(parser);
```

Comment:
I'm wondering if it would be possible/worthwhile to augment GetResult (and its parsing method) with an optional exception field and then only use GetResult.fromXContentEmbedded() for both cases. That would add something to GetResult that is unused in the normal Get-API, but it would simplify things here. I'm on the fence about this idea, maybe @javanna can comment on this too.